### ISSUE TITLE
[Meta] Suggest issues deserving bounties

### ISSUE THREAD

**Issue by TheFrenchGhosty** [iv-org/invidious#2199]
This is a meta issue where organizations members, and users in general can suggest issues that deserve to have bounties added to them.

**Comment by syeopite** [comment]
Basically any issues that involves a massive rewrite or overhaul should have an bounty imo.

Like #1951. I mean I'll personally do it for free but It'll be nice to have some sort of momentary compensation lol.

Anyhow,
Any *major* features of Invidious that are broken should also have one. Like #1979 or the whole lsquic fiasco. 
Any *major* features Youtube has but Invidious doesn't *and* would be difficult to implement should also have one. #438, #92, #879 etc. 

And maybe we should also bump the existing bounties since no one is tackling them.

**Comment by SamantazFox** [comment]
What @syeopite said, plus maybe #2098, #1762 and #1561, maybe?

**Comment by TheFrenchGhosty** [comment]
@syeopite 

>Any major features of Invidious that are broken should also have one. Like #1979 or the whole lsquic fiasco.

Bounty added on #1979. Thank you :)

About lsquic I really don't know what has to be done, so I will let you guys handle it to be sure it's "complete".

>Any major features that needs or should be added and would be difficult to implement should also have one. Add Livechat on Livestreams #438, Circumventing geo-restrictions #92, How does decentralization / federation work and how to improve it? #879 etc.

Waiting for @Perflyst 

Since we started adding bounties Perflyst wanted us to only add them on bugs (and not enhancements/feature requests). Back then it was mainly because there was so much problems with Invidious, but now I guess it's time to add some bounties to enhancements/feature requests.

**Comment by TheFrenchGhosty** [comment]
@SamantazFox 

> #2098

We have to discuss if we really want Google SSO on Invidious, in my opinion it's a pretty bad thing to have on a service mainly made to avoid Google.

> #1762

If it's only an issue when `decrypt_polling` is enabled... is it really a problem? According to [the description you added on your config.example.yml rewrite](https://github.com/iv-org/invidious/blob/2e54312cccf5bcdf94f01edcab98780c4af18b64/config/config.example.yml#L333) it allows access to "secured videos", I really don't understand what this mean, can you please explain?

> #1561

I know what PubSub is however I'm wondering... is it really useful to have? I think we had this discussion before and if I remember correctly the answer was yes, but I would just like to be sure.

**Comment by Perflyst** [comment]
> or the whole lsquic fiasco

If someone can define what needs to be done I guess we can add a bounty. I can't estimate how much work this will be.

> Add Livechat on Livestreams #438 

Even though it is my own feature request I still think this feature is not very important compared to other bugs / improvements left
We can put some bounties on most wanted feature requests I guess

> Circumventing geo-restrictions #92

Sounds good to me, go for it

> How does decentralization / federation work and how to improve it? #879 

This looks more like a discussion to me than some feature request

**Comment by SamantazFox** [comment]
> We have to discuss if we really want Google SSO on Invidious, in my opinion it's a pretty bad thing to have on a service mainly made to avoid Google.

Some people want to use their G-account to continue to interact with the content creators/community (comments/likes) without being too much tracked and annoyed by Youtube itself. And if we work better on accessibly, it could be another reason (than privacy) why people choose us.

> If it's only an issue when `decrypt_polling` is enabled... is it really a problem? According to [the description you added on your config.example.yml rewrite](https://github.com/iv-org/invidious/blob/2e54312cccf5bcdf94f01edcab98780c4af18b64/config/config.example.yml#L333) it allows access to "secured videos", I really don't understand what this mean, can you please explain?

Some videos, mostly music, contain a form of DRM that requires a small algorithm to run on the client side to "verify" that they can play the video. decrypt_polling is a "job" (a function ran periodically by crystal) that extracts this algorithm. Because this function is currently broken, issues like #1947 happen.

> I know what PubSub is however I'm wondering... is it really useful to have? I think we had this discussion before and if I remember correctly the answer was yes, but I would just like to be sure.

I honnestly have no clue about what PubSub is, lol. But from what I remember from "hovering the code", it's required for subscriptions or something?

**Comment by unixfox** [comment]
> I know what PubSub is however I'm wondering... is it really useful to have? I think we had this discussion before and if I remember correctly the answer was yes, but I would just like to be sure.

Yes it is very important, for big instances it allows to instantly get notified once a new video comes out instead of waiting 5 days that Invidious refresh all the 100 000 channels in the database.

**Comment by jtagcat** [comment]
If the budget is not all booked, allocating it to items in todos might be of interest. Though, where is the limit — 2$ bounties for minor bugs, papercuts, feel wrong.

**Comment by lunainvictum** [comment]
Hello folks,

This is probably going to really hurt what I'm about to say, but I've been following this project for a few years now. And since the main developer left this project, I've read some comments here from "high ranking members" who said it would be difficult to program for Invidious, because the possible developer base is very small, and the ones who know crystal is even smaller.

So I wonder what it would cost to completely reprogram Invidious from scratch? With a platform independent and widely used programming language like python, c, c++, c#, java or what ever? Which one would be the best, I honestly wouldn't know. I once programmed something like Invidious for fun in PHP and Python and youtube-dl + ffmpeg. But that was only very rudimentary and required that you enter in advance in the database which youtube channels you want to watch. 

I would appreciate it if one would simply discuss it and if one has found a programming language or has agreed, simply starts an attempt and starts a crowdfunding on indiegogo or kickstarter or other platforms. And then spread this all over the social media (not aggressively, of course, but purely informatively). 

So maybe invidious could be helped to be what it should be? a really solid, privacy-friendly frontend alternative to youtube. 

And if it's a programming language "that most people know" (and I'm just saying that many people who use invidious are people who are already tech-savvy and might know a programming language that's not as exotic as crystal), then invidious could of course grow much better and faster, and respond to youtube's tyranny.

Just my 2 cents and an idea. Sry, this whole text is translated from german to english with deepl. But it seems correct.

**Comment by SamantazFox** [comment]
> because the possible developer base is very small, and the ones who know crystal is even smaller.

Crystal is extremely similar to ruby, and honnestly, if you can program in Ruby, you can program in Crystal too. It just requires to read a bit the documentation for specifics, and that's about it.

> So I wonder what it would cost to completely reprogram Invidious from scratch? With a platform independent and widely used programming language like python, c, c++, c#, java or what ever?

There are/were some attempts to rewrite some parts of invidious in python (e.g: [Invidious-quic-proxy](https://github.com/iv-org/Invidious-quic-proxy)). C/C++ is just a big nope, given how string manipulation is complex and prone to errors (We're using a lot of JSON, and we already have enought memory leaks to manage with a "memory safe" language). And IIRC we already discussed about Java, but I don't recall much enthusiasm about it.

> And if it's a programming language "that most people know" ([...] a programming language that's not as exotic as crystal), then invidious could of course grow much better and faster

As I said, Crystal has a lot in common with Ruby ^^ My guess on why there aren't much Crystal devs would be that it's a name they have never heard of, a language with no associated "hype wave" (unlike Node.js or Rust, for instance), so people are less prone to dig in that language, compared to others. 

> and respond to youtube's tyranny.

Don't worry for that part! We're doing team work and info sharing with the nice peeps at NewPipe and youtube-dl :)

**Comment by SapeurPonpier** [comment]
I think #2236  deserves a bounty. Using invidious on iOS has become tedious.

**Comment by TheFrenchGhosty** [comment]
@SapeurPonpier Thank you for the suggestion, bounty added!

**Comment by ChunkyProgrammer** [comment]
I think the automatic db migration could be a good candidate for a bounty. It's blocking a few PRs https://github.com/iv-org/invidious/pull/2254#issuecomment-1650707286

**Comment by syeopite** [comment]
I think we should consider putting up bounties for whoever can discover the source(s) of the memory leaks in Invidious

**Comment by TheFrenchGhosty** [comment]
@syeopite 

> I think we should consider putting up bounties for whoever can discover the source(s) of the memory leaks in Invidious

Sure, we should discuss it
